src/main/java/gyro/aws/secretsmanager/SecretResource.java

```diff
@@ -0,0 +1,456 @@
+    }
+
+    public void setRotationEnabled(Boolean rotationEnabled) {
+        this.rotationEnabled = rotationEnabled;
+    }
+
+    /**
+     * The unique ARN of a Lambda function that's invoked by Secrets Manager to rotate the secret either automatically
+     * per the schedule or manually by a call to RotateSecret.
+     */
+    @Output
+    public String getRotationLambdaARN() {
+        return rotationLambdaARN;
+    }
+
+    public void setRotationLambdaARN(String rotationLambdaARN) {
+        this.rotationLambdaARN = rotationLambdaARN;
+    }
+
+    /**
+     * The structure that contains the rotation configuration for the secret.
+     */
+    @Output
+    public SecretRotationRulesType getSecretRotationRulesType() {
+        return secretRotationRulesType;
+    }
+
+    public void setSecretRotationRulesType(SecretRotationRulesType secretRotationRulesType) {
+        this.secretRotationRulesType = secretRotationRulesType;
+    }
+
+    /**
+     * The unique identifier associated with the version of the secret.
+     */
+    @Output
+    public String getVersionId() {
+        return versionId;
+    }
+
+    public void setVersionId(String versionId) {
+        this.versionId = versionId;
+    }
+
+    /**
+     * The list of all of the currently assigned VersionStage staging labels and the VersionId that each is attached
+     * to.
+     */
+    @Output
+    public Map<String, List<String>> getVersionIdsToStages() {
+        return versionIdsToStages;
+    }
+
+    public void setVersionIdsToStages(Map<String, List<String>> versionIdsToStages) {
+        this.versionIdsToStages = versionIdsToStages;
+    }
+
+    @Override
+    public boolean refresh() {
+        SecretsManagerClient client = createClient(SecretsManagerClient.class);
+        DescribeSecretResponse response = null;
+
+        try {
+            response = client.describeSecret(r -> r.secretId(getArn()));
+        } catch (ResourceNotFoundException ex) {
+            // No Resource found
+        }
+
+        if (response == null) {
+            return false;
+        }
+
+        copyFrom(response);
+        return true;
+    }
+
+    @Override
+    public void create(GyroUI ui, State state) throws Exception {
+        SecretsManagerClient client = createClient(SecretsManagerClient.class);
+
+        CreateSecretResponse response = client.createSecret(r ->
+            r.description(getDescription())
+            .kmsKeyId(getKmsKey() != null ? getKmsKey().getArn() : null)
+            .name(getName())
+            .secretBinary(getSecretBinary() != null ? SdkBytes.fromUtf8String(getSecretBinary()) : null)
+            .secretString(getSecretString())
+            .tags(convertTags(getTags()))
+        );
+
+        DescribeSecretResponse secret = null;
+
+        try {
+            secret = client.describeSecret(r -> r.secretId(response.arn()));
+        } catch (ResourceNotFoundException ex) {
+            // No Resource found
+        }
+        copyFrom(secret);
```

Comment:
You are doing essentially what refresh does without using the returns
```suggestion
        setArn(response.arn());
        refresh();
```